homeassistant/components/sensorpush/sensor.py

```diff
@@ -0,0 +1,148 @@
+"""Support for sensorpush ble sensors."""
+from __future__ import annotations
+
+from typing import Optional, Union
+
+from sensorpush_ble import (
+    ATTR_MANUFACTURER as SENSOR_MANUFACTURER,
+    ATTR_MODEL as SENSOR_MODEL,
+    ATTR_NAME as SENSOR_NAME,
+    DeviceClass,
+    DeviceKey,
+    SensorDeviceInfo,
+    SensorUpdate,
+)
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothCoordinatorEntity,
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothDataUpdateCoordinator,
+    PassiveBluetoothEntityKey,
+)
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    PERCENTAGE,
+    PRESSURE_PA,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+SENSOR_DESCRIPTIONS = {
+    (DeviceClass.TEMPERATURE, "°C"): SensorEntityDescription(
+        key=f"{DeviceClass.TEMPERATURE}_°C",
+        name=DeviceClass.TEMPERATURE.name.replace("_", " ").title(),
```

Comment:
Having the name in the description isn't going to work since we could have 10 temp sensors on the device.  It needs to move to entity data